src/localize-router.service.ts

```diff
@@ -73,8 +100,18 @@ export class LocalizeRouterService {
    */
   private parseSegmentValue(snapshot: ActivatedRouteSnapshot): string {
     if (snapshot.routeConfig) {
-      let subPathSegments = snapshot.routeConfig.path.split('/');
-      return subPathSegments.map((s: string, i: number) => s.indexOf(':') === 0 ? snapshot.url[i].path : s).join('/');
+      if (snapshot.routeConfig.path === '**') {
+        let urls = [];
+        for (let subPathSegment of snapshot.url) {
```

Comment:
Please use `.filter` and `.map` on `snapshot.url` to simplify code e.g.:

snapshot.url.filter(segment => segment.path).map(segment => segment.path).join('/');